pkg/webui/account/views/create-account/index.js

```diff
@@ -91,151 +87,124 @@ const getSuccessMessage = state => {
   }
 }
 
-@withRouter
-@connect(
-  state => ({
-    fetching: state.user.fetching,
-    user: selectUser(state),
-  }),
-  {
-    push,
-    replace,
-  },
-)
-export default class CreateAccount extends React.PureComponent {
-  static propTypes = {
-    fetching: PropTypes.bool.isRequired,
-    push: PropTypes.func.isRequired,
-    user: PropTypes.user,
+const CreateAccount = ({ location }) => {
+  const [fetching, isConfigError] = useRequest(getIsConfiguration())
+  if (Boolean(isConfigError)) {
+    throw isConfigError
   }
 
-  static defaultProps = {
-    user: undefined,
-  }
-
-  constructor(props) {
-    super(props)
-    this.state = {
-      error: '',
-    }
-  }
-
-  @bind
-  async handleSubmit(values, { setSubmitting, setErrors }) {
-    try {
-      const { user_id, ...rest } = values
-      const { invitation_token = '' } = queryString.parse(location.search)
-      const { push } = this.props
-      const result = await api.users.register({
-        user: { ids: { user_id }, ...rest },
-        invitation_token,
-      })
-
-      push(`/login${location.search}`, {
-        info: getSuccessMessage(result.data.state),
-      })
-    } catch (error) {
-      this.setState({
-        error: error.response.data,
-      })
-    } finally {
-      setSubmitting(false)
-    }
+  const [error, setError] = useState(undefined)
+  const passwordRequirements = useSelector(selectPasswordRequirements)
```

Comment:
I can but it would need an extra selector to extract the password requirements from the is config object (and not from the root of the store). I don't think it's worth the hassle.